client/client.go

```diff
@@ -574,6 +574,56 @@ func (r *NotaryRepository) GetDelegationRoles() ([]*data.Role, error) {
 	return allDelegations, nil
 }
 
+// RoleWithSignatures is a Role with its associated signatures
+type RoleWithSignatures struct {
+	Signatures []data.Signature
+	data.Role
+}
+
+// GetRepoRoleMetaInfo returns a list of RoleWithSignatures objects for this repo
+// This represents the latest metadata for each role in this repo
+func (r *NotaryRepository) GetRepoRoleMetaInfo() ([]RoleWithSignatures, error) {
```

Comment:
I think a method that returns the whole state of the repository in a well delineated object is preferable, exactly because of what @cyli said.

I would prefer people calling ListTargets when they merely want to get all the targets (probably the most common occurence), and have another method to get a snapshot of the current repository with all the info that it provides.